### PR TITLE
[MachineVerifier] Improve G_EXTRACT_SUBVECTOR checking

### PR DESCRIPTION
**Pull request by topperc**
Check that the destination of G_EXTRACT_SUBVECTOR is smaller than the source.
Fix wording of error messages

### CODE REVIEW

File: llvm/lib/CodeGen/MachineVerifier.cpp [hunk 1]
```diff
@@ -1793,8 +1799,7 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
     }
 
     uint64_t SrcMinLen = SrcTy.getElementCount().getKnownMinValue();
-    if (SrcTy.isScalable() == DstTy.isScalable() &&
-        (Idx >= SrcMinLen || Idx + DstMinLen > SrcMinLen)) {
+    if ((Idx >= SrcMinLen || Idx + DstMinLen > SrcMinLen)) {
```
Comment (michaelmaitland) [inline llvm/lib/CodeGen/MachineVerifier.cpp:1802]: I think you can drop `()` pair

File: llvm/test/MachineVerifier/test_g_extract_subvector.mir [hunk 2]
```diff
@@ -19,35 +20,35 @@ body:             |
...
 
     %14:_(<2 x s32>) = G_IMPLICIT_DEF
     %15:_(<4 x s32>) = G_IMPLICIT_DEF
 
-    ; CHECK: Source type and index must not cause extract to overrun to the destination type
+    ; CHECK: Destination type and index must not cause extract to overrun the source vector
     %16:_(<2 x s32>) = G_EXTRACT_SUBVECTOR  %14, 4
 
-    ; CHECK: Source type and index must not cause extract to overrun to the destination type
+    ; CHECK: Destination type and index must not cause extract to overrun the source vector
```
Comment (michaelmaitland) [inline llvm/test/MachineVerifier/test_g_extract_subvector.mir:51]: How can we overrun the source vector if we are writing to the destination vector? Overrun usually occurs when you write too much data to too small of a buffer.

Comment (topperc) [reply llvm/test/MachineVerifier/test_g_extract_subvector.mir:51]: The overrun is reading off the end of the source vector. The number of elements to read is determined by the destination, so you can't write too many elements to it.

Comment (michaelmaitland) [reply llvm/test/MachineVerifier/test_g_extract_subvector.mir:51]: I see

**Review by michaelmaitland — APPROVED** [review]
LGTM

**Pull request merged**